weather_dl/download_pipeline/pipeline.py

```diff
@@ -166,9 +167,15 @@ def run(argv: t.List[str], save_main_session: bool = True) -> PipelineArgs:
     num_requesters_per_key = known_args.num_requests_per_key
     client = CLIENTS[client_name](config)
     if num_requesters_per_key == -1:
-        num_requesters_per_key = client.num_requests_per_key(
-            config.dataset
-        )
+        if client_name == 'eumetsat':
+            num_requesters_per_key = client.num_requests_per_key(
+                config.dataset,
+                bool(config.selection.get('eumetsat_format_conversion_to'))
+            )
+        else:
+            num_requesters_per_key = client.num_requests_per_key(
+                config.dataset
+            )
```

Comment:
This shouldn't be necessary. We already delegate calculating the number of requests per key by the download class. Instead of adding an if statement, let's see if we can add this logic without creating a special case for this client. Refactoring the general case is OK.